src/GraphQL/Validation/ValidationError.cs

```diff
@@ -12,6 +12,24 @@ public class ValidationError : DocumentError
     {
         private readonly List<ASTNode> _nodes = new();
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ValidationError"/> class with a specified error message.
+        /// </summary>
+        public ValidationError(string message) : base(message)
+        {
+            Code = GetValidationErrorCode(GetType());
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ValidationError"/> class with a specified error message. Sets the
+        /// <see cref="ExecutionError.Code">Code</see> property based on the inner exception.
+        /// Loads any exception data from the inner exception into this instance.
+        /// </summary>
+        public ValidationError(string message, Exception? innerException) : base(message, innerException)
+        {
+            Code = GetValidationErrorCode(GetType());
```

Comment:
> Code property based on the inner exception

You override code with one from current exception.

Reply:
This is copied from the existing constructors here and I believe it is valid.  The idea is to return the Code that points to the current derived of the validation error class -- for example an InvalidVariableError always returns INVALID_VALUE regardless of what inner exception caused the value conversion problem.  However Codes can still be examined to see the the inner exception code.  (Compared to UnhandledError which would always give the code for the underlying exception.)

Reply:
> Sets additional codes based on the inner exception(s).

This remark no longer belongs to this class, it is better to remove it. No more `Codes` property here. `ErrorInfoProvider` as a default implementation of `IErrorInfoProvider` reads codes in `GetCodesForError` and then stores this info into `extensions`.

Reply:
While that is true, it helps the user to understand what the effective result of passing an exception to the constructor is.  That is why the comment exists that way in all of these classes.  Any changes to the comments should retain this concept, even if it is written as a remark (for example, "Inner exceptions are typically used to write the 'codes' property to the error's extension property.").

While I personally think it's fine, feel free to issue a separate PR to change the language of the xml comments throughout the source.  As of now this comment will retain the same language as the rest of the execution error class constructors.

Reply:
I will clarify this a bit.

Reply:
Well, I don't know what to do.  It could use clarification.  I'm leaving it.  Feel free to make a PR.

Reply:
The xml comment in DocumentError needs to be fixed too.

Reply:
ok